script/mafia

```diff
@@ -26,16 +26,23 @@ run_upgrade () {
 
   trap clean_up EXIT
 
+  MAFIA_CUR="$0"
+
+  if [ -L "$MAFIA_CUR" ]; then
+    echo 'Refusing to overwrite a symlink; run `upgrade` from the canonical path.' >&2
```

Comment:
Tangential, anyone know a non-horrible portable way to dereference a symlink in shell? Best I can come up with is parsing `stat(1)` output.